server/sonar-server/src/main/java/org/sonar/server/view/index/ViewIndexer.java

```diff
@@ -129,4 +133,37 @@ private void clearLookupCache(String viewUuid) {
     }
   }
 
+  @Override
+  public IndexingResult index(DbSession dbSession, Collection<EsQueueDto> items) {
+    if (items.isEmpty()) {
+      return new IndexingResult();
+    }
+
+    Set<String> views = items
+      .stream()
+      .map(EsQueueDto::getDocId)
+      .collect(toHashSet(items.size()));
+
+    BulkIndexer bulkIndexer = newBulkIndexer(Size.REGULAR, new OneToOneResilientIndexingListener(dbClient, dbSession, items));
+    bulkIndexer.start();
+
+    // Safety check to remove all views that may not have been deleted
+    views.removeAll(dbClient.componentDao().selectExistingUuids(dbSession, views));
```

Comment:
It would be better to really index the views referenced by `dbClient.componentDao().selectExistingUuids(dbSession, views)`. Nevertheless for the time being a warning in the javadoc of this method should highlight the fact that views are NOT indexed (because no real usage).